static/app/views/performance/trends/changeExplorerUtils/functionsList.tsx

```diff
@@ -0,0 +1,440 @@
+      percentChange: percentageChange,
+      avgTimeDifference:
+        functionAfter.avgSumExclusiveTime - (functionBefore?.avgSumExclusiveTime || 0),
+      changeType:
+        percentageChange < 0 ? FunctionChangeType.improved : FunctionChangeType.regressed,
+    };
+  });
+}
+
+export function NumberedFunctionsList(props: NumberedFunctionsListProps) {
+  const {project, transactionName, organization, limit, isLoading, isError, functions} =
+    props;
+
+  if (isLoading) {
+    return <LoadingIndicator />;
+  }
+
+  if (isError) {
+    return (
+      <ErrorWrapper>
+        <IconWarning
+          data-test-id="error-indicator-functions"
+          color="gray200"
+          size="xxl"
+        />
+        <p>{t('There was an issue finding suspect functions for this transaction')}</p>
+      </ErrorWrapper>
+    );
+  }
+
+  if (functions?.length === 0 || !functions) {
+    return (
+      <EmptyStateWarning>
+        <p data-test-id="functions-no-results">
+          {t('No results found for suspect functions')}
+        </p>
+      </EmptyStateWarning>
+    );
+  }
+
+  // percent change of a function must be more than 1%
+  const formattedFunctions = functions
+    ?.filter(func => (functions.length > 10 ? Math.abs(func.percentChange) >= 1 : true))
+    .slice(0, limit)
+    .map((func, index) => {
+      const profiles = func['examples()'] as string[];
+
+      const functionSummaryView = generateProfileFlamechartRouteWithQuery({
+        orgSlug: organization.slug,
+        projectSlug: project?.slug || '',
+        profileId: profiles[0],
+        query: {
+          frameName: func.function as string,
+          framePackage: func.package as string,
+        },
+      });
+
+      const handleClickAnalytics = () => {
+        trackAnalytics(
+          'performance_views.performance_change_explorer.function_link_clicked',
+          {
+            organization,
+            transaction: transactionName,
+            package: func.package as string,
+            function: func.function as string,
+            profile_id: profiles[0],
+          }
+        );
+      };
+
+      return (
+        <li key={`list-item-${index}`}>
+          <ListItemWrapper data-test-id="list-item">
+            <p style={{marginLeft: space(2)}}>
+              {tct('[changeType] suspect function', {changeType: func.changeType})}
+            </p>
+            <ListLink to={functionSummaryView} onClick={handleClickAnalytics}>
+              {func.function}
+            </ListLink>
+          </ListItemWrapper>
+        </li>
+      );
+    });
+
+  if (formattedFunctions?.length === 0) {
+    return (
+      <EmptyStateWarning>
+        <p data-test-id="functions-no-changes">
+          {t('No sizable changes in suspect functions')}
+        </p>
+      </EmptyStateWarning>
+    );
+  }
+
+  return (
+    <div style={{marginTop: space(4)}}>
+      <h6>{t('Relevant Suspect Functions')}</h6>
+      <ol>{formattedFunctions}</ol>
+    </div>
```

Comment:
I think we should always display the header even when things are loading so we know what's trying to load in. Instead of having the div and header here, can we move this up to where `NumberedFunctionsList` is being called so all of the loading and content is rendered as children?

Reply:
ahh yes good call out. I'll fix that up :)